pkg/api/customization/pipeline/pipeline.go

```diff
@@ -85,7 +126,33 @@ func (h *Handler) run(apiContext *types.APIContext) error {
 	if err != nil {
 		return err
 	}
-	execution, err := utils.GenerateExecution(h.Pipelines, h.PipelineExecutions, pipeline, utils.TriggerTypeUser)
+	runPipelineInput := v3.RunPipelineInput{}
+	requestBytes, err := ioutil.ReadAll(apiContext.Request.Body)
+	if err != nil {
+		return err
+	}
+	if string(requestBytes) != "" {
+		if err := json.Unmarshal(requestBytes, &runPipelineInput); err != nil {
+			return err
+		}
+	}
+	if len(pipeline.Spec.Stages) < 1 ||
+		len(pipeline.Spec.Stages[0].Steps) < 1 ||
+		pipeline.Spec.Stages[0].Steps[0].SourceCodeConfig == nil {
```

Comment:
The above condition is the same as Validatitor func, what about putting it in a new func and call it in both places:
```
func validDefinition(pipeline)  bool {
if len(pipeline.Spec.Stages) < 1 ||
		len(pipeline.Spec.Stages[0].Steps) < 1 ||
 		pipeline.Spec.Stages[0].Steps[0].SourceCodeConfig == nil {
   return false
}
return true
}
```

Reply:
updated